packages/back-end/src/util/organization.util.ts

```diff
@@ -50,6 +52,50 @@ export const ALL_PERMISSIONS = [
   ...ENV_SCOPED_PERMISSIONS,
 ];
 
+export function getUserPermissions(userId: string, org: OrganizationInterface) {
+  const roles = getRoles(org);
+  const rolePermissionsMap: Record<string, Set<Permission>> = {};
+
+  roles.forEach((role) => {
+    rolePermissionsMap[role.id] = new Set(role.permissions);
+  });
+
+  const memberInfo = org.members.find((m) => m.id === userId);
```

Comment:
Good call - I like the idea of returning early with an empty `UserPermissions` object, rather than throwing an error. Upstream from this there are blocks that handle the error of not having permissions to do something, and throwing an error causes the app to break when we're trying to get the UserContext object and results in a perpetual loading state.